module/zfs/dsl_prop.c

```diff
@@ -1230,6 +1230,18 @@ dsl_prop_get_all_ds(dsl_dataset_t *ds, nvlist_t **nvp,
 			goto out;
 	}
 
+#ifdef __FreeBSD__
+		nvlist_t *propval;
+		dsl_dataset_name(ds, setpoint);
+		VERIFY0(nvlist_alloc(&propval, NV_UNIQUE_NAME, KM_SLEEP));
+		VERIFY0(nvlist_add_string(propval, ZPROP_VALUE,
+		    (ds->ds_jailname && INGLOBALZONE(curproc)) ?
+		    ds->ds_jailname : "0"));
+		VERIFY0(nvlist_add_string(propval, ZPROP_SOURCE, setpoint));
+		VERIFY0(nvlist_add_nvlist(*nvp, "jail", propval));
```

Comment:
The `fnvlist_*` wrappers should be used here, or better yet add the required error handling even if they're almost certain never to fail.

Reply:
Sure. It seems I followed the examples from `dsl_prop_get_all_impl()` itself. Migrated to `fnvlist_*`, thanks.